packages/flutter/test/material/raised_button_test.dart

```diff
@@ -617,6 +617,45 @@ void main() {
     expect(box.size, equals(const Size(76, 36)));
     expect(childRect, equals(const Rect.fromLTRB(372.0, 293.0, 428.0, 307.0)));
   });
+
+  testWidgets('RaisedButton.icon responds to applied padding', (WidgetTester tester) async {
+    const Key buttonKey = Key('test');
+    const Key labelKey = Key('label');
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: Center(
+            child: RaisedButton.icon(
+              icon: const Icon(Icons.add),
+              padding: const EdgeInsets.all(16),
```

Comment:
I just thought of this, but maybe using different padding values for every edge would be a safer to ensure that the padding gets applied to the correct edges.
```suggestion
              padding: const EdgeInsets.fromLTRB(5.0, 6.0, 7.0, 8.0),
```